pkg/codegen/go/gen.go

```diff
@@ -2562,12 +2592,30 @@ func (pkg *pkgContext) getTypeImports(t schema.Type, recurse bool, importsAndAli
 		return
 	}
 	seen[t] = struct{}{}
+
+	// Import an external type with `token` and return true.
+	// If the type is not external, return false.
+	importExternal := func(token string) bool {
+		if pkg.isExternalReference(t) {
+			extPkgCtx := pkg.contextForExternalReference(t)
+			mod := extPkgCtx.tokenToPackage(token)
+			imp := path.Join(extPkgCtx.importBasePath, mod)
+			importsAndAliases[imp] = extPkgCtx.pkgImportAliases[imp]
+			return true
+		}
+		return false
+	}
+
 	switch t := t.(type) {
 	case *schema.OptionalType:
 		pkg.getTypeImports(t.ElementType, recurse, importsAndAliases, seen)
 	case *schema.InputType:
 		pkg.getTypeImports(t.ElementType, recurse, importsAndAliases, seen)
 	case *schema.EnumType:
+		if importExternal(t.Token) {
```

Comment:
Isn't the line immediately below it try to handle externals also?

```
mod := pkg.tokenToPackage(t.Token)
if mod != pkg.mod  // here
```

I wonder if that is now redundant (dead code)?

Reply:
It shouldn't be. Notice that we are comparing `mod` instead of `pkg`. This is us handling enums from other modules (but the same pkg).

Reply:
Ah thanks.